test.js

```diff
@@ -30,8 +30,9 @@ test('unist-util-map', function (t) {
   t.end()
 
   /**
-   * @param {Node} node
-   * @returns {Node}
+   * @template {Node} TNode
+   * @param {TNode} node
+   * @returns {TNode}
```

Comment:
a full name may help for the generic, maybe something like `NodeType` or `NodeLike`

```suggestion
   * @template {Node} NodeType
   * @param {NodeType} node
   * @returns {NodeType}
```


```suggestion
   * @template {Node} NodeLike
   * @param {NodeLike} node
   * @returns {NodeLike}
```